web/src/main/java/com/bazaarvoice/emodb/web/throttling/DataStoreUpdateThrottleControlTask.java

```diff
@@ -0,0 +1,154 @@
+ *
+ * Whether per API key or instance wide all rate limits are expressed and enforced at the instance level.  There is no way
+ * to specify a global rate limit across the entire cluster.  If an API key is rate limited to 200 wps and there are 4
+ * instances in the cluster then, assuming the API key is maximally utilizing all four instances, it can achieve an
+ * aggregate 800 wps across the cluster.  If a fifth server is added to the cluster its maximum throughput rises to
+ * 1,000 wps.
+ *
+ * Throttles are applied only within the local data center.  For example, rate limiting an API key in us-east-1 will
+ * have no effect on throttling writes in eu-west-1.  This task would have to be called on a server in each data center
+ * to rate limit the API key in both.  However, any updates made using this task are automatically propagated to every
+ * server in the local data center.
+ *
+ * All throttles must expire after a specific duration.  If none is provided the default is 24 hours.  To remove all
+ * throttling for an API key set the rate limit to 0.  Note that it is not possible to completely block update access
+ * using this task, but it can be set remarkably low, such as to 0.1 (1 update every 10 seconds).
+ *
+ * API keys are expressed using their IDs, not the private keys.  Using this task with a key and not the key's ID will
+ * not return an error but it will also not rate limit any API keys.  To throttle updates instance wide across all API keys
+ * use the wildcard API key, "*".
+ *
+ * Usage:
+ * <pre>
+ *     # Set a throttle for a single API key, with the default and with an explicit expiration
+ *     curl -s -XPOST 'localhost:8081/tasks/sor-api-update-throttle?id=ID01&limit=100'
+ *     curl -s -XPOST 'localhost:8081/tasks/sor-api-update-throttle?id=ID02&limit=100&duration=PT1H'
+ *
+ *     # Set an instance wide throttle
+ *     curl -s -XPOST 'localhost:8081/tasks/sor-api-update-throttle?id=*&limit=500&duration=PT30M'
+ *
+ *     # Remove a throttle
+ *      curl -s -XPOST 'localhost:8081/tasks/sor-api-update-throttle?id=ID02&limit=0'
+ *
+ *     # List all throttled endpoints
+ *      curl -s -XPOST 'localhost:8081/tasks/sor-api-update-throttle'
+ *
+ *     # Clear all throttled endpoints
+ *     curl -s -XPOST 'localhost:8081/tasks/sor-api-update-throttle?clear'
+ * </pre>
+ */
+public class DataStoreUpdateThrottleControlTask extends Task {
+    private final DataStoreUpdateThrottleManager _throttleManager;
+
+    @Inject
+    public DataStoreUpdateThrottleControlTask(TaskRegistry taskRegistry,
+                                              DataStoreUpdateThrottleManager throttleManager) {
+        super("sor-api-update-throttle");
+        _throttleManager = requireNonNull(throttleManager, "throttleManager");
+        taskRegistry.addTask(this);
+    }
+
+    @Override
+    public void execute(ImmutableMultimap<String, String> parameters, PrintWriter out) throws Exception {
```

Comment:
The `throws Exception` seems to be unnecessary.

Reply:
Meh, it's from the `Task` abstract parent class:

`public abstract void execute(ImmutableMultimap<String, String> var1, PrintWriter var2) throws Exception;`